doc/SAI-Proposal-VlanStack-QinQ.md

```diff
@@ -0,0 +1,466 @@
+
+Add support for Vlan Stack
+-------------------------------------------------------------------------------
+
+ Title       | Vlan Stack
+-------------|-----------------------------------------------------------------
+ Authors     | Edge-Core
+ Status      | In review
+ Type        | Standards track
+ Created     | 25/02/2022
+ Updated     | -
+ SAI-Version | 1.9.1
+
+-------------------------------------------------------------------------------
+
+## Overview ##
+
+The Vlan Stack feature includes two functions, vlan translation and QinQ.
+
+![qinq](figures/vlan_stack/vlan_stack_QinQ_usecase.png "Figure 1: QinQ usage scenario")
+__Figure 1: QinQ usage scenario__
+
+![vlan_translate](figures/vlan_stack/vlan_stack_vlan_xlate_usecase.png "Figure 2: Vlan translation usage scenario")
+__Figure 2: Vlan Translation usage scenario__
```

Comment:
If L3 is enabled (ex: ip addr configured) on the translated VLAN and if the incoming packets are copied to CPU, what would be the VLAN on the packet? is it the CVLAN or the Translated VLAN?

Reply:
In some cases, the packets are copied to CPU before handling by the switch chip. For those packets, the VLAN will be the one before translated.
But this may be different for other chip vendors.

Reply:
Should we define the expected behavior as part of this proposal? I understand that some of the older switching chips may not be compliant with the SAI requirements.

Reply:
I think this PR already defined the expected behavior of the vlan header when the 802.1ad rule was set.
All packets transferred in Layer 2 should follow the rule.
It is another thing how the switch chip handle copy-to-CPU packets technically.